### PR TITLE
Update changelog generation script

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

* Fix regression from #11652
* Fix [MD022](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md032---lists-should-be-surrounded-by-blank-lines) / [MD032](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md032---lists-should-be-surrounded-by-blank-lines) rule violations
* Modify `Get-ChangeLog` to generate the changelog according to the format from #11652.

## PR Context

Follow-up to #11652

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by xtqqczze** [comment]
@TravisEz13 Since this functionality requires an additional parameter `ThisReleaseTag` I have avoided making changes to `Get-ChangeLog`. If you would like to to make changes, you could push to this branch on xtqqczze/PowerShell-PowerShell.

File: tools/releaseTools.psm1 [hunk 1]
```diff
@@ -341,9 +341,34 @@ function Get-ChangeLog
     PrintChangeLog -clSection $clDocs -sectionTitle 'Documentation and Help Content'
 }
 
+function Get-ChangeLogEx
```
Comment (TravisEz13) [inline tools/releaseTools.psm1]: Can we build this functionality into `Get-ChangeLog` . (feel free to rename the existing function)

File: tools/releaseTools.psm1 [hunk 2]
```diff
@@ -341,9 +341,34 @@ function Get-ChangeLog
...
+
+    Get-ChangeLog -LastReleaseTag:$LastReleaseTag -Token:$Token -HasCherryPick:$HasCherryPick
+
+    Write-Output "[${thisRelease}]: https://github.com/PowerShell/PowerShell/compare/${LastReleaseTag}...${ThisReleaseTag}`n"
+}
+
 function PrintChangeLog($clSection, $sectionTitle) {
     if ($clSection.Count -gt 0) {
-        "### $sectionTitle"
+        "### $sectionTitle`n"
```
Comment (TravisEz13) [inline tools/releaseTools.psm1]: Thanks

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

**Comment by TravisEz13** [comment]
@adityapatwardhan Should be port this to 7.0?

File: tools/releaseTools.psm1 (hunk 1 above)
Comment (xtqqczze) [reply tools/releaseTools.psm1]: Yes we can, this will change the function parameters.

Comment (xtqqczze) [reply tools/releaseTools.psm1]: I have made new `$ThisReleaseTag` parameter optional.

File: tools/releaseTools.psm1 [hunk 3]
```diff
@@ -137,9 +137,12 @@ function New-CommitNode
 function Get-ChangeLog
 {
     param(
-        [Parameter(Mandatory)]
+        [Parameter(Mandatory = $true)]
         [string]$LastReleaseTag,
 
+        [Parameter(Mandatory = $false)]
```
Comment (TravisEz13) [inline tools/releaseTools.psm1]: @adityapatwardhan 
I think we should make the mandatory?

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
@PoshChan Please remind me in 4 hours

**Comment by PoshChan** [comment]
@TravisEz13, this is the reminder you requested 4 hours ago

File: tools/releaseTools.psm1 (hunk 3 above)
Comment (adityapatwardhan) [reply tools/releaseTools.psm1]: I agree, this should be mandatory

Comment (TravisEz13) [reply tools/releaseTools.psm1]:
```suggestion
        [Parameter(Mandatory)]
```

File: tools/releaseTools.psm1 [hunk 4]
```diff
@@ -328,6 +331,21 @@ function Get-ChangeLog
...
+    $base = $LastReleaseTag
+
+    if ($PSBoundParameters.ContainsKey('ThisReleaseTag')) {
+        $head = $ThisReleaseTag
+        $version = $ThisReleaseTag.TrimStart('v')
+    }
+    else {
+        $head = $(git rev-parse HEAD)
+        $version = 'Unreleased'
+    }
```
Comment (TravisEz13) [inline tools/releaseTools.psm1]: Please update since the parameter is mandatory

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

**Comment by xtqqczze** [comment]
PR is ready to merge.

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
@msftbot status

**Comment by xtqqczze** [comment]
@TravisEz13 Any further improvements to the changelog generation script you think could be made?

**Comment by TravisEz13** [comment]
Personally, I think the build and packaging section, should just list the PR numbers and not the description (perhaps thank any non-microsoft contributor).
I like this a lot more now that you have links to the full history.

**Comment by TravisEz13** [comment]
Or better yet list it as details, for example 

<details>
<summary>11736</summary>
Update changelog generation
</details>


```
<details>
<summary>11736</summary>
Update changelog generation
</details>
```

**Pull request merged**